### PR TITLE
Patch 3

### PR DESCRIPTION
**Pull request by sethulekshmi**

### CODE REVIEW

**Review by aparnaprem — APPROVED** [review]

File: Technical Design/blockchain-datastructure.txt [hunk 1]
```diff
@@ -0,0 +1,203 @@
+
+{
+    "hash":"0000000000000bae09a7a393a8acded75aa67e46cb81f7acaa5ad94f9eacd103",
+    "ver":1,
+    "prev_block":"00000000000007d0f98d9edca880a6c124e25095712df8952e0439ac7409738a",
+    "mrkl_root":"935aa0ed2e29a4b81e0c995c39e06995ecce7ddbebb26ed32d550a72e8200bf5",
+    "time":var json {“coordinates”:”2°”:”24᾽”:”12.2᾽᾽” , “time”:”2᾽”:”24᾽᾽”},
```
Comment (nikhilkrishna) [inline Technical Design/blockchain-datastructure.txt]: What is the mrkl_root containing ? Is it only 1 transaction hash ? Are we storing multiple transactions in a single block ? @sethulekshmi

Comment (nikhilkrishna) [reply Technical Design/blockchain-datastructure.txt]: Please correct this. 
Time should be of standard format and in UTC - yyyy-MM-dd HH:mm:ss (https://www.w3.org/TR/NOTE-datetime)

If you are looking to add GPS co-ordinates that should either be an additional field or modification of the "location" field. They also should be in standard format - degrees, minutes, seconds lattitude and longitude.

File: Technical Design/blockchain-datastructure.txt [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+    "n_tx":22,
+    "size":9195,
+    "block_index":818041,
+    "main_chain":true,
+    "height":154595,
+    "received_time":1322131301,
+    "relayed_by":"108.60.208.156",
+    "tx":[current value:5d44,
+          owner1:miner,
+          owner2:shop keeper]
```
Comment (nikhilkrishna) [inline Technical Design/blockchain-datastructure.txt]: please make the transaction fields more clear. I cannot tell from this who is the buyer and who is the seller. Perhaps you can use the field "from" and "to" for indicating how the ownership is transferred. 
Also we need to have the unique identifier for the owners here. This should be a unique canonical value.

Comment (aparnaprem) [reply Technical Design/blockchain-datastructure.txt]: ok sir.we made certain changes. replaced the transaction array with 'from' and 'to'. also changed the time and date format.

File: Technical Design/blockchain-datastructure.txt (hunk 1 above)
Comment (sethulekshmi) [reply Technical Design/blockchain-datastructure.txt]: mrkl_root contains the hash of the first role that is miner and it remains same.A particular block stores only one transaction,

**Pull request merged**